### ISSUE TITLE
Auth flow, declarative vs changenotifier

### ISSUE THREAD

**Issue by Silfalion** [Milad-Akarie/auto_route_library#597]
Hey, there! Thanks for this repo.

Noticed from the other issues that using a declarative auto_router for auth flow wasn't mentioned anywhere beside the documentations. Is there any drawback to it?

The approach I went to implement is to redirect the user on login, that's simple, but then redirect him to the route on disconnect.

**Comment by ArneSaknussemm89** [comment]
There are a couple drawbacks I experienced when using declarative without digging in too deep:

There's really isn't a way to "pop" back, since it's controlled by the returned list of routes. I.e. when I made a declarative flow and allowed a user to go "back" simply by changing the step they were on, the page transition looked like they were going "forward". (The old page slid in from Right -> Left instead of from Left -> Right)

Redirecting back to the "flow" is also easy if you just push that Route from anywhere else. The declarative approach worked nicely using `riverpod` as well as all I needed was a StateNotifier for the step and then use a Provider so the widget was rebuilt when the state changed.
